lib/Backends/CPU/DebugInfo.cpp

```diff
@@ -356,6 +375,10 @@ void LLVMIRGen::generateDebugInfo() {
     // If a function has a debug information already, no need to re-emit it.
     if (F.getSubprogram())
       continue;
+    // Specializations are not forced to have any debug info. They will have it,
```

Comment:
Why do we need this?  Do we inject the debug info both before and after specialization?

Reply:
Specializations are supposed to have the debug info only if the original function had it. This will happen automatically during the specialization when cloning the original function. There is a comment in the specializer about it.

Reply:
This while loop with a bunch of checks is not necessary anymore. I removed it altogether.